app/lib/email_alert_list_title_builder.rb

```diff
@@ -0,0 +1,62 @@
+class EmailAlertListTitleBuilder
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(filter:, subscription_list_title_prefix:, facets:)
+    @filter = filter
+    @subscription_list_title_prefix = subscription_list_title_prefix
+    @facets = facets
+  end
+
+  def call
+    join_fragment = "in the following #{'category'.pluralize(selected_facet_choice_names.count)}"
+    "#{subscription_list_title_prefix.strip} #{join_fragment}: #{comma_separated_facet_choices}"
+  end
+
+private
+
+  attr_accessor :filter, :subscription_list_title_prefix, :facets
+
+  def comma_separated_facet_choices
+    selected_facet_choice_names.map { |choice_name| "'#{choice_name}'" }.join(", ")
+  end
+
+  def selected_facet_choice_names
+    @selected_facet_choice_names ||= begin
+      overwritten_filters = filter.map do |selected_facet_id, selected_facet_choices|
+        selected_facet_choices.map do |selected_facet_choice_id|
+          fetch_facet_choice_name(selected_facet_id, selected_facet_choice_id)
+        end
+      end
+      overwritten_filters.flatten
+    end
+  end
+
+  def fetch_facet_choice_name(facet_id, facet_choice_id)
+    fetch_facet_choice_name_from_overrides(facet_id, facet_choice_id) || fetch_facet_choice_name_from_data(facet_id, facet_choice_id)
+  end
+
+  def fetch_facet_choice_name_from_overrides(facet_id, facet_choice_id)
+    facet_name_overrides.fetch(facet_id, {})[facet_choice_id]
+  end
+
+  def fetch_facet_choice_name_from_data(selected_facet_id, selected_facet_choice_key)
+    facet_info = facets.detect { |facet| facet["facet_id"] == selected_facet_id }
+    facet_choice = facet_info["facet_choices"].detect { |choice| choice["key"] == selected_facet_choice_key }
+    facet_choice["topic_name"]
+  end
+
+  def facet_name_overrides
```

Comment:
We did consider that, but felt it would make it harder to delete later on. On the assumption that these overrides won't be necessary forever, and even that this class won't be necessary forever, we can easily delete the snowflakey code here as it's in one place.

If we integrate it properly into the fixture data / content item metadata, it's more coupled into the application as a whole and therefore harder to remove.

It would make _this_ class much simpler, but add potential technical debt to a few other places. On balance we chose not to. Do let me know if you disagree!

Reply:
I see.

Originally, I thought these values would be best placed in https://github.com/alphagov/search-api/blob/master/config/find-eu-exit-guidance-business-email-signup.yml, however this would require `govuk_content_schemas` to be updated.

As you say, this would involve a more substantial process when it comes to removing this finder.

Reply:
👍 in that case, are you happy to approve this?